src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -591,13 +591,18 @@ public void write(int series)
           pathName, datasetAttributes, gridPosition
         ).toByteBuffer();
 
-        int length = blockWidth * bytesPerPixel;
+        int destLength = blockWidth * bytesPerPixel;
+        int srcLength = destLength;
+        if (fileType == FileType.zarr) {
+          // n5/n5-zarr does not de-pad on read
+          srcLength = activeTileWidth;
```

Comment:
Not sure I understand - why doesn't this need to be multiplied by ```bytesPerPixel```?

Reply:
Oversight; good catch.